xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp

```diff
@@ -116,8 +116,9 @@ void CGUIDialogPVRTimerSettings::SetTimer(CFileItem *item)
   m_strTitle      = m_timerInfoTag->m_strTitle;
 
   const CDateTime now(CDateTime::GetCurrentDateTime());
-  m_startLocalTime    = m_timerInfoTag->IsStartAtAnyTime() ? now : InitializeDateTime(m_timerInfoTag->StartAsLocalTime());
-  m_endLocalTime      = m_timerInfoTag->IsEndAtAnyTime()   ? now : InitializeDateTime(m_timerInfoTag->EndAsLocalTime());
+  bool bFirstDay((m_timerType && m_timerType->SupportsFirstDay()) || (GetDateAsInt(m_timerInfoTag->FirstDayAsUTC()) != 0));
```

Comment:
Please do not use <code>GetDateAsInt</code> here. The fact that it returns 0 for time_t(0) is only a coincidence. The semantics of GetDateAsInt is to obtain a unique index value for the GUI spinners, no more no less. I was already warned by a code reviewer about the bad naming of that method. I will change the name soon to reflect it's actual purpose.

What you actually need is <code>CTimerInfoTag::IsFirstDayAtAnyTime</code>, which does not yet exist. But it should be a no brainer for you to implement it, right.